tools/OMIcheck/OMIcheck.ps1

```diff
@@ -1,127 +1,460 @@
-﻿# This script lists the virtual machines in Azure subscriptions of the user,
+﻿# This script lists the virtual machines and scale sets in Azure subscriptions of the user,
 # and looks for vulnerable OMI packages. The report is grouped by subscription and resource group name.
 
 # Set below flag to true if you want vulnerable OMI to be patched.
-# If OMSLinuxAgent is installed, then a minor update will be triggered.
-# If OMI is standalone, the latest bits will be installed from Github.
-$upgradeOMI = $false; #detect only
-$omsVersionGood = "1.13"
+#    - If OMSLinuxAgent or LinuxDiagnostics is installed, then a minor update will be triggered.
+#    - OMI package is installed from Github.
+$upgradeOMI = $false #detect only
+
+$OmiServerGoodVersion = "OMI-1.6.8-1"
+$OmiPkgGoodVersion = "1.6.8.1"
+$LadPkgGoodVersion = "1.5.110-LADmaster.1483"
```

Comment:
This version of mdsd shipped in both LAD 3.0.131 and LAD 3.0.135. The presence of this package version wouldn't tell us whether or not LAD needs to be updated.

Reply:
The VM extension also lacks the patch number. Therefore, I update the OMI package regardless.